src/hb-machinery.hh

```diff
@@ -35,6 +35,9 @@
 #include "hb-iter.hh"
 #include "hb-vector.hh"
 
+#if defined(__x86_64) || defined(__ARM_NEON__) || defined(__ARM_NEON)
```

Comment:
Is unaligned access safe on NEON?

This needs a comment, why we only do it for some archs.  Also, since the condition is duplicated multiple times, we should just define a symbol like HB_USE_BSWAP and condition on that.  Like:
```
#ifndef HB_USE_BSWAP
#  if ...
#    define HB_USE_BSWAP 1
#  else
#    define HB_USE_BSWAP 0
#endif

later...

#if HB_USE_BSWAP
  return ...
#endif
```